x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/cluster/metadata/MetadataMigrateToDataTiersRoutingServiceTests.java

```diff
@@ -0,0 +1,533 @@
+            List<String> migratedIndices = migrateIndices(mb, state, "data");
+            assertThat(migratedIndices.size(), is(0));
+
+            ClusterState migratedState = ClusterState.builder(ClusterName.DEFAULT).metadata(mb).build();
+            IndexMetadata migratedIndex = migratedState.metadata().index("indexWithBoxAttribute");
+            assertThat(migratedIndex.getSettings().get(DATA_ROUTING_REQUIRE_SETTING), nullValue());
+            assertThat(migratedIndex.getSettings().get(BOX_ROUTING_REQUIRE_SETTING), is("warm"));
+            assertThat(migratedIndex.getSettings().get(INDEX_ROUTING_PREFER), nullValue());
+        }
+
+        {
+            IndexMetadata.Builder indexNoRoutingAttribute =
+                IndexMetadata.builder("indexNoRoutingAttribute").settings(getBaseIndexSettings());
+
+            ClusterState state =
+                ClusterState.builder(ClusterName.DEFAULT).metadata(Metadata.builder().put(indexNoRoutingAttribute)).build();
+
+            Metadata.Builder mb = Metadata.builder(state.metadata());
+            List<String> migratedIndices = migrateIndices(mb, state, "data");
+            assertThat(migratedIndices.size(), is(0));
+
+            ClusterState migratedState = ClusterState.builder(ClusterName.DEFAULT).metadata(mb).build();
+            IndexMetadata migratedIndex = migratedState.metadata().index("indexNoRoutingAttribute");
+            assertThat(migratedIndex.getSettings().get(DATA_ROUTING_REQUIRE_SETTING), nullValue());
+            assertThat(migratedIndex.getSettings().get(BOX_ROUTING_REQUIRE_SETTING), nullValue());
+            assertThat(migratedIndex.getSettings().get(INDEX_ROUTING_PREFER), nullValue());
+        }
+    }
+
+    public void testRequireAttributeIndexSettingTakesPriorityOverInclude() {
+        IndexMetadata.Builder indexWithRequireAndInclude =
+            IndexMetadata.builder("indexWithRequireAndInclude")
+                .settings(getBaseIndexSettings()
+                    .put(DATA_ROUTING_REQUIRE_SETTING, "warm")
+                    .put(DATA_ROUTING_INCLUDE_SETTING, "cold")
+                );
+        ClusterState state =
+            ClusterState.builder(ClusterName.DEFAULT).metadata(Metadata.builder().put(indexWithRequireAndInclude)).build();
+
+        Metadata.Builder mb = Metadata.builder(state.metadata());
+
+        List<String> migratedIndices = migrateIndices(mb, state, "data");
+        assertThat(migratedIndices.size(), is(1));
+        assertThat(migratedIndices.get(0), is("indexWithRequireAndInclude"));
+
+        ClusterState migratedState = ClusterState.builder(ClusterName.DEFAULT).metadata(mb).build();
+        IndexMetadata migratedIndex = migratedState.metadata().index("indexWithRequireAndInclude");
+        assertThat("index migration ONLY clears the setting it migrates, in this case the require.data setting",
+            migratedIndex.getSettings().get(DATA_ROUTING_INCLUDE_SETTING), is("cold"));
```

Comment:
I think we'll want to clear both of these, and pick one to favor (as I mentioned above), what do you think?